graylog2-server/src/main/java/org/graylog2/indexer/IndexSetValidator.java

```diff
@@ -17,26 +17,65 @@
+import org.graylog2.indexer.rotation.strategies.TimeBasedRotationStrategyConfig;
+import org.graylog2.plugin.indexer.retention.RetentionStrategyConfig;
+import org.graylog2.shared.utilities.StringUtils;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
 import org.joda.time.Duration;
+import org.joda.time.Period;
 
+import javax.annotation.Nullable;
 import javax.inject.Inject;
 import java.util.Optional;
 
 public class IndexSetValidator {
     private final static Duration MINIMUM_FIELD_TYPE_REFRESH_INTERVAL = Duration.standardSeconds(1L);
     private final IndexSetRegistry indexSetRegistry;
+    private final ElasticsearchConfiguration elasticsearchConfiguration;
 
     @Inject
-    public IndexSetValidator(IndexSetRegistry indexSetRegistry) {
+    public IndexSetValidator(IndexSetRegistry indexSetRegistry, ElasticsearchConfiguration elasticsearchConfiguration) {
         this.indexSetRegistry = indexSetRegistry;
+        this.elasticsearchConfiguration = elasticsearchConfiguration;
     }
 
     public Optional<Violation> validate(IndexSetConfig newConfig) {
+
+        // Don't validate prefix conflicts in case of an update
+        if (Strings.isNullOrEmpty(newConfig.id())) {
+            final Violation prefixViolation = validatePrefix(newConfig);
+            if (prefixViolation != null) {
+                return Optional.of(prefixViolation);
+            }
+        }
+
+        final Violation refreshIntervalViolation = validateRefreshInterval(newConfig);
+        if (refreshIntervalViolation != null) {
+            return Optional.of(refreshIntervalViolation);
+        }
+
+        return Optional.ofNullable(validateRetentionPeriod(newConfig));
+    }
+
+    @Nullable
+    private Violation validateRefreshInterval(IndexSetConfig newConfig) {
+        // Ensure fieldTypeRefreshInterval is not shorter than a second, as that may impact performance
+        if (newConfig.fieldTypeRefreshInterval().isShorterThan(MINIMUM_FIELD_TYPE_REFRESH_INTERVAL)) {
+            return Violation.create("Index field_type_refresh_interval \"" + newConfig.fieldTypeRefreshInterval().toString() + "\" is too short. It must be 1 second or longer.");
+        }
+        return null;
+    }
+
+    @Nullable
+    private Violation validatePrefix(IndexSetConfig newConfig) {
         // Build an example index name with the new prefix and check if this would be managed by an existing index set
         final String indexName = newConfig.indexPrefix() + MongoIndexSet.SEPARATOR + "0";
         if (indexSetRegistry.isManagedIndex(indexName)) {
-            return Optional.of(Violation.create("Index prefix \"" + newConfig.indexPrefix() + "\" would conflict with an existing index set!"));
+            return Violation.create("Index prefix \"" + newConfig.indexPrefix() + "\" would conflict with an existing index set!");
         }
 
         // Check if an existing index set has a more generic index prefix.
```

Comment:
This is only half the story. Should be something like:
// Check if the new index set configuration has a more generic index prefix than an existing index set, or vice versa.

Reply:
Good catch. I didn't write that comment but I'll fix it with your suggestion.